src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/HMAC.cs

```diff
@@ -17,13 +19,14 @@ protected int BlockSizeValue
 
         protected HMAC() { }
 
-        public static new HMAC Create() =>
+        public static new HMAC? Create() =>
             throw new PlatformNotSupportedException(SR.Cryptography_DefaultAlgorithm_NotSupported);
 
-        public static new HMAC Create(string algorithmName) =>
-            (HMAC)CryptoConfigForwarder.CreateFromName(algorithmName);
+        public static new HMAC? Create(string algorithmName) =>
+            (HMAC?)CryptoConfigForwarder.CreateFromName(algorithmName);
 
-        public string HashName
+        [DisallowNull]
+        public string? HashName
```

Comment:
(Repeat from the ref) This should probably be considered non-nullable, it's up to the derived types to set this property in their ctor; any type which doesn't has a bug.

Reply:
`[DissalowNull]` which restricts `null` being set is not enough? The intent is getter shouldn't return null?

Reply:
Yeah, users of the class should never get a `null` value back from get_HashName; and if they get one it's a bug.